src/main/java/com/example/chapter1/basicprogramming/BinarySearchRecursively.java

```diff
@@ -0,0 +1,25 @@
+package com.example.chapter1.basicprogramming;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.util.Arrays;
+
+public class BinarySearchRecursively {
+
+
+    public static int rank(int[] input, int searchElement) {
+        if (ArrayUtils.isEmpty(input)) throw new IllegalArgumentException("Empty array.");
```

Comment:
same for `sort` ?
Keep in mind, we also have this method inside `Collections.sort()`.
Which way is appropriate ? 
Should we statically import static methods which are having same name inside another class ? 
Also by java conventions we should use class name while accessing its static member.
So i think its kind of debate which one should use where.